src/test.js

```diff
@@ -692,47 +696,112 @@ export function test( testName, callback ) {
 	newTest.queue();
 }
 
-extend( test, {
-	todo: function todo( testName, callback ) {
-		if ( focused || config.currentModule.ignored ) {
-			return;
-		}
+function todo( testName, data, callback ) {
+	if ( focused || config.currentModule.ignored ) {
+		return;
+	}
 
-		const newTest = new Test( {
-			testName,
-			callback,
-			todo: true
-		} );
+	const newTest = new Test( {
+		testName,
+		callback,
+		todo: true,
+		params: data
+	} );
 
-		newTest.queue();
-	},
-	skip: function skip( testName ) {
-		if ( focused || config.currentModule.ignored ) {
-			return;
-		}
+	newTest.queue();
+}
 
-		const test = new Test( {
-			testName: testName,
-			skip: true
-		} );
+function skip( testName ) {
+	if ( focused || config.currentModule.ignored ) {
+		return;
+	}
+
+	const test = new Test( {
+		testName: testName,
+		skip: true
+	} );
+
+	test.queue();
+}
+function only( testName, data, callback ) {
+	if ( config.currentModule.ignored ) {
+		return;
+	}
+	if ( !focused ) {
+		config.queue.length = 0;
+		focused = true;
+	}
 
-		test.queue();
+	const newTest = new Test( {
+		testName: testName,
+		callback: callback,
+		params: data
+	} );
+
+	newTest.queue();
+}
+
+function normalizeTable( data ) {
+	const isTable = data.every( Array.isArray );
+	if ( isTable ) {
+		return data;
+	}
+	return data.map( datum => [ datum ] );
+}
+
+function runEach( data, eachFn ) {
+	if ( Array.isArray( data ) ) {
+		normalizeTable( data ).forEach( eachFn );
+	} else {
+		throw new Error(
+			`test.each expects array of arrays or an array of primitives is the expected input.
```

Comment:
"primitives _as_ the expected input." ?

Reply:
Changed to: "... expects _an_ array of arrays or an array of primitives _as_ ...". Thanks for bringing this up.